### PR TITLE
Added some extension functions

### PR DESCRIPTION
**Pull request by Gokuldroid**

### CODE REVIEW

File: kandroid/build.gradle [hunk 1]
```diff
@@ -31,7 +31,7 @@ def gitUrl = 'https://github.com/pawegio/KAndroid.git'
 
 android {
     compileSdkVersion 25
-    buildToolsVersion "25.0.2"
+    buildToolsVersion "25.0.3"
```
Comment (pawegio) [inline kandroid/build.gradle]: Why you want to decrease build tools version?

Comment (illuzor) [reply kandroid/build.gradle]: I agree. Latest version is 26.0.1

Comment (Gokuldroid) [reply kandroid/build.gradle]: I have modified to the same version as in your repo in final commit

Comment (pawegio) [reply kandroid/build.gradle]: Ok, I see.

File: kandroid/src/main/kotlin/com/pawegio/kandroid/KApi.kt [hunk 2]
```diff
@@ -24,4 +24,12 @@ inline fun toApi(toVersion: Int, inclusive: Boolean = false, action: () -> Unit)
...
 inline fun fromApi(fromVersion: Int, inclusive: Boolean = true, action: () -> Unit) {
     if (Build.VERSION.SDK_INT > fromVersion || (inclusive && Build.VERSION.SDK_INT == fromVersion)) action()
+}
+
+inline fun fromLollipopAbove(action: () -> Unit) {
+    fromApi(Build.VERSION_CODES.LOLLIPOP, true, action)
+}
+
+inline fun fromMarshMellowAbove(action: () -> Unit) {
+    fromApi(Build.VERSION_CODES.M, true, action)
```
Comment (pawegio) [inline kandroid/src/main/kotlin/com/pawegio/kandroid/KApi.kt:34]: I'm not sure about these two functions. @illuzor what do you think?

File: kandroid/src/main/kotlin/com/pawegio/kandroid/KActivity.kt [hunk 3]
```diff
@@ -49,4 +52,14 @@ inline fun Activity.lockCurrentScreenOrientation() {
...
+}
+
+inline fun Activity.keepScreenOn() {
+    window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+}
+
+inline fun SupportActivity.replaceFragment(@IdRes id: Int, fragment: SupportFragment) {
+    val manager = supportFragmentManager.beginTransaction()
+    manager.replace(id, fragment)
+    manager.commit()
```
Comment (pawegio) [inline kandroid/src/main/kotlin/com/pawegio/kandroid/KActivity.kt:64]: you can use `run()` extension here as follows:
```
supportFragmentManager.beginTransaction().run {
   replace(id, fragment)
   commit()
}
```

File: kandroid/src/main/kotlin/com/pawegio/kandroid/KFragment.kt [hunk 4]
```diff
@@ -45,4 +48,12 @@ inline fun SupportFragment.toast(@StringRes resId: Int): Unit = activity.toast(r
...
-inline fun <reified T : Preference> PreferenceFragment.findPref(key: String): T = findPreference(key) as T
\ No newline at end of file
+inline fun <reified T : Preference> PreferenceFragment.findPref(key: String): T = findPreference(key) as T
+
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+fun Fragment.getHostingActivity(): Activity? {
+    if (activity != null) {
+        return activity
+    }
+    return parentFragment?.getHostingActivity()
```
Comment (pawegio) [inline kandroid/src/main/kotlin/com/pawegio/kandroid/KFragment.kt:58]: 1. Simplify it with elvis operator: `return activity ?: parentFragment?.getHostingActivity()`
2. Convert it to `val`

**Comment by pawegio** [comment]
Code review finished. Thanks for your contribution! 👍

File: kandroid/src/main/kotlin/com/pawegio/kandroid/KApi.kt (hunk 2 above)
Comment (Gokuldroid) [reply kandroid/src/main/kotlin/com/pawegio/kandroid/KApi.kt:34]: Lot of things were supported from lollipop and above  API versions. Like elevation of UI elements. It is just to make more readable.

Comment (illuzor) [reply kandroid/src/main/kotlin/com/pawegio/kandroid/KApi.kt:34]: I think these functions are useless. fromApi()/toApi are very simple to use and flexible.

Comment (pawegio) [reply kandroid/src/main/kotlin/com/pawegio/kandroid/KApi.kt:34]: Yes, it depends on project so better approach is to extract and reuse such functions internally in android app module (when API from specific versions is used more often).

**Pull request closed without merging**